File: c_solutions_11-15/Euler_14.c
#include <stdio.h>
#include <time.h>


int main(void) 
{
    clock_t start, stop;
    start = clock();

    int high, count;
    long long high_num, canidate, collatz;

    for (canidate=3, high=0; canidate < 1000000; canidate += 2, count=0) {
        for (collatz=canidate; collatz != 1; count++) 
            collatz = collatz % 2 == 0 ? collatz / 2 : (collatz * 3) + 1; 
        if (count > high) {
            high = count;
            high_num = canidate;
        }
    }

    printf ("Answer: %lld\n", high_num);
    stop = clock();
    printf ("Time: %f\n", ((float)stop - (float)start) / CLOCKS_PER_SEC);

    return 0;
}
        
        

File: perl_solutions_11-20/Euler_14.pl
use strict;
use warnings;
use Time::HiRes qw( clock );


sub collatz
{
    my ($cnt, $high, $nhigh) = (0, 0, 0);
    foreach my $i (map(($_ * 2) + 1, (1..500000))) {
        my $cur = $i;
        while ($i != 1) {
            if ($i % 2 != 0) {
                $i *= 3;
                $i += 1;
                $cnt += 1;
            }
            if ($i % 2 == 0) {
                $i /= 2;
                $cnt += 1;
            }
        }
        if ($cnt > $high) {
            $high = $cnt;
            $nhigh = $cur;
        }
        $cnt = 0;
    }
    return $nhigh;
}

my $start = clock();
my $answer = collatz();
my $stop = clock();
print "Answer: $answer\n";
printf "Time: %.5f\n", $stop - $start;
